### PR TITLE
[DDW-536] Service agreement can be skipped by accessing to Settings

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR disabled certain menu items when accepting terms of use.
[Jira Ticket](https://jira.iohk.io/browse/DDW-536)

## Todos
- [x] Disable `Settings`, `Wallet Settings`, `About Daedalus`, and `Daedalus Diagnostics` menu items when accepting terms of use

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1610959942016900)

### App
- [x] Start daedalus and make sure the menu items mentioned above are disabled when accepting terms of use

### Test Cases 

**Scenario 1 - Validate "Wallet settings" is disabled on the "Initial Settings" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Initial Settings" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Wallet settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 2 - Validate "Settings" is disabled is disabled on "Initial Settings" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Initial Settings" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 3 - Validate "About" is disabled is disabled on "Initial Settings" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Initial Settings" screen
When I click on "Daedalus"
Then I can see the "About" option is disabled
```

**Scenario 4 - Validate "Daedalus Diagnostics" is disabled is disabled on "Initial Settings" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Initial Settings" screen
When I click on "Help"
Then I can see the "Daedalus Diagnostics" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 5 - Validate "Wallet settings" is disabled on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Wallet settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 6 - Validate "Settings" is disabled is disabled on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 7 - Validate "About" is disabled is disabled on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I click on "Daedalus"
Then I can see the "About" option is disabled
```

**Scenario 8 - Validate "Daedalus diagnostics" is disabled is disabled on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I click on "Help"
Then I can see the "Daedalus Diagnostics" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 9 - Validate "Wallet settings" is enabled after agreeing on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I have accepted the terms on "Service agreement" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Wallet settings" option is enabled
And I can access by the keyboard shortcut
```

**Scenario 10 - Validate "Settings" is enabled after agreeing on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I have accepted the terms on "Service agreement" screen
When I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Settings" option is enabled
And I can access by the keyboard shortcut
```

**Scenario 11 - Validate "About" is enabled after agreeing on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I have accepted the terms on "Service agreement" screen
When I click on "Daedalus"
Then I can see the "About" option is enabled
```

**Scenario 12 - Validate "Daedalus Diagnostics" is enabled after agreeing on "Service Agreement" screen**
```Gherkin
Given Daedalus is executed for the first time after installation
And I have accepted the terms on "Service agreement" screen
When I click on "Help"
Then I can see the "Daedalus Diagnostics" option is enabled
And I can access by the keyboard shortcut
```
**Scenario 13 - Validate "Wallet Settings" is disabled after submitting initial settings and restarting Daedalus**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I restart Daedalus
And I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Wallet Settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 14 - Validate "Settings" is disabled after submitting initial settings and restarting Daedalus**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I restart Daedalus
And I click on "View" (Or "Daedalus" on macOS)
Then I can see the "Settings" option is disabled
And I can't access by the keyboard shortcut
```

**Scenario 15 - Validate "About" is disabled after submitting initial settings and restarting Daedalus**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I restart Daedalus
And I click on "Daedalus"
Then I can see the "About" option is disabled
```

**Scenario 16 - Validate "Daedalus Diagnostics" is disabled after submitting initial settings and restarting Daedalus**
```Gherkin
Given Daedalus is executed for the first time after installation
And I am on the "Service agreement" screen
When I restart Daedalus
And I click on "Help"
Then I can see the "Daedalus Diagnostics" option is disabled
And I can't access by the keyboard shortcut
```

## Testing Summary

English
image
image
image
image

Japanese
image
image
image
image

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov. Tested on build 15928. Works really well when on the Service Level Agreement screen. However Service Level Agreement can be skipped as these options are available still from the initial settings screen

|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1 | Fail| Wallet Settings is still an available option from the initial settings screen and is not greyed out.
|Scenario 2 | Fail| Settings is still an available option from the initial settings screen. This option can be used as a way of bypassing the Service Level Agreement.<a href="https://drive.google.com/file/d/19UJC8TYyuT_d9E2niLGTFl36HNbgEGv8/view?usp=sharing">video</a>  |
|Scenario 3 | Fail| About Daedalus options is still available from the initial settings screen This can also be used as a way to skip the service level agreement

**Comment by yakovkaravelov** [comment]
> Hi @yakovkaravelov. Tested on build 15928. Works really well when on the Service Level Agreement screen. However Service Level Agreement can be skipped with scenario 1(If a wallet is saved in state directory from before) and 2 as these options are available still from the initial settings screen
> 
> Test cases	Results	Note
> Scenario 1	Fail	Wallet Settings is still an available option from the initial settings screen and is not greyed out.
> Scenario 2	Fail	Settings is still an available option from the initial settings screen. This option can be used as a way of bypassing the Service Level Agreement.[video](https://drive.google.com/file/d/19UJC8TYyuT_d9E2niLGTFl36HNbgEGv8/view?usp=sharing)
> Scenario 3	Fail	About Daedalus options is still available from the initial settings screen

@ManusMcCole This was fixed. Please test again, thanks.

**Review by ManusMcCole — APPROVED** [review]
Goodjob Yakov. Looks good 👍

**Comment by gnpf** [comment]
The test cases were updated to validate "Daedalus Diagnostics" (cases 4, 8, 12) and also to validate all the options after submitting the initial settings and restarting Daedalus (cases 13 to 16).

@yakovkaravelov On build 15934 the cases 1 to 12 passed (pending to implement 13 to 16).

**Comment by yakovkaravelov** [comment]
@gnpf @nikolaglumac This is read for review/test.
Thanks.

File: source/renderer/app/stores/ProfileStore.js [hunk 1]
```diff
@@ -302,10 +304,16 @@ export default class ProfileStore extends Store {
...
     await this.getTermsOfUseAcceptanceRequest.execute();
+    await enableTermsAgreedUserMenuItemsChannel.send();
   };
 
-  _getTermsOfUseAcceptance = () => {
-    this.getTermsOfUseAcceptanceRequest.execute();
+  _getTermsOfUseAcceptance = async () => {
+    await this.getTermsOfUseAcceptanceRequest.execute();
+    if (this.getTermsOfUseAcceptanceRequest.result) {
+      await enableTermsAgreedUserMenuItemsChannel.send();
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ProfileStore.js:313]: @yakovkaravelov please rename this channel to: `enableApplicationMenuNavigationChannel`

File: source/renderer/app/stores/ProfileStore.js [hunk 2]
```diff
@@ -302,10 +304,16 @@ export default class ProfileStore extends Store {
...
+    } else {
+      await disableTermsAgreedUserMenuItemsChannel.send();
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ProfileStore.js:315]: @yakovkaravelov please rename this channel to: `disableApplicationMenuNavigationChannel`

File: source/main/index.js [hunk 3]
```diff
@@ -202,14 +204,40 @@ const onAppReady = async () => {
     await safeExit();
   });
 
-  buildAppMenus(mainWindow, cardanoNode, locale, { isUpdateAvailable: false });
+  buildAppMenus(mainWindow, cardanoNode, locale, {
+    isUpdateAvailable: false,
+    isOnInitialSettings: false,
```
Comment (nikolaglumac) [inline source/main/index.js:209]: @yakovkaravelov please rename to `isNavigationEnabled`

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@yakovkaravelov please see my comments. Thanks!

**Comment by yakovkaravelov** [comment]
@nikolaglumac They were renamed, thanks.

File: source/main/index.js [hunk 4]
```diff
@@ -206,26 +206,26 @@ const onAppReady = async () => {
 
   buildAppMenus(mainWindow, cardanoNode, locale, {
     isUpdateAvailable: false,
-    isOnInitialSettings: false,
+    isNavigationEnabled: true,
```
Comment (nikolaglumac) [inline source/main/index.js:209]: @yakovkaravelov don't you think we should disable the navigation by default?

**Review by gnpf — APPROVED** [review]

File: source/main/index.js (hunk 4 above)
Comment (yakovkaravelov) [reply source/main/index.js]: @nikolaglumac Fixed.

**Comment by nikolaglumac** [comment]
@ManusMcCole please re-test this one 🙏

**Review by ManusMcCole — APPROVED** [review]
👍

File: source/renderer/app/stores/ProfileStore.js [hunk 5]
```diff
@@ -302,10 +304,16 @@ export default class ProfileStore extends Store {
...
+    await enableApplicationMenuNavigationChannel.send();
   };
 
-  _getTermsOfUseAcceptance = () => {
-    this.getTermsOfUseAcceptanceRequest.execute();
+  _getTermsOfUseAcceptance = async () => {
+    await this.getTermsOfUseAcceptanceRequest.execute();
+    if (this.getTermsOfUseAcceptanceRequest.result) {
+      await enableApplicationMenuNavigationChannel.send();
+    } else {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ProfileStore.js:314]: @yakovkaravelov I don't think we need this else clause as the menu navigation is disabled by default?

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@yakovkaravelov nice work!
Now with the latest changes, I believe we actually don't even need the "disable application menu channel". The navigation is disabled by default and we only need to enable it either when user accepts the terms of use OR if we detect the terms of use have already been accepted in the past.

File: source/main/index.js [hunk 6]
```diff
@@ -202,14 +203,29 @@ const onAppReady = async () => {
...
+      })
+  );
 
   await rebuildApplicationMenu.onReceive(
     (data) =>
       new Promise((resolve) => {
         locale = getLocale(network);
         buildAppMenus(mainWindow, cardanoNode, locale, {
           isUpdateAvailable: data.isUpdateAvailable,
+          isNavigationEnabled: true,
```
Comment (nikolaglumac) [inline source/main/index.js:228]: @yakovkaravelov this should not be always set to true

**Review by nikolaglumac — CHANGES_REQUESTED** [review]

File: source/main/index.js (hunk 6 above)
Comment (yakovkaravelov) [reply source/main/index.js]: @nikolaglumac This is fixed. Thanks.

**Review by nikolaglumac — APPROVED** [review]
Looks and works great now @yakovkaravelov 🎉

**Pull request merged**